File: 0205.isomorphic_strings/solution.java
class Solution{
    public boolean isIsomorphic(String s, String t){
        int [] m1 = new int[256];
        int [] m2 = new int[256];
        // 存两个字符串对应索引的值
        for (int i=0;i<s.length();i++){
            if (m1[s.charAt(i)] != m2[t.charAt(i)]){
                return false;
            }
            m1[s.charAt(i)] = m2[s.chatAt(i)] = i+1;
        }
        return true;
    }
}

File: 0205.isomorphic_strings/solution.py
class Solution:
    def is_isomorphic(self, s: str, t: str) -> bool:
        """ 根据值在两个字符串中的索引是否相同 """
        for i, v in enumerate(s):
            if s.find(v) != t.find(t[i]):
                return False
        return True

    def other_solution(self, s, t):
        """ 将s中的字符对应的t中的字符存入字典，如果不符合条件则为False """
        d = {}
        for i, v in enumerate(s):
            if d.get(v):
                if d[v] != t[i]:
                    return False
            elif t[i] in d.values():
                return False
            d[v] = t[i]
        return True


if __name__ == "__main__":
    print(Solution().test("abc", "bac"))
